web/packages/teleport/src/UnifiedResources/ResourceActionButton.tsx

```diff
@@ -425,3 +409,130 @@ const makeDesktopLoginOptions = (
+ * If launchUrls is empty, ButtonWithMenu is returned with the href value
+ * configured with samlAppSsoUrl.
+ * @param samlAppSsoUrl - ACS URL (also known as SSO endpoint) value for SAML app.
+ * @param launchUrls - custom service provider endpoints that will be used to initiate
+ *                     authentication, instead of the ACS URL.
+ * @param SamlActionMenu - SAML app edit and delete action menu items.
+ */
+function makeSamlAppLoginWithMenuButton(
+  samlAppSsoUrl: string,
+  launchUrls: SamlAppLaunchUrl[],
+  SamlActionMenu: JSX.Element
+) {
+  let ssoUrl = samlAppSsoUrl;
+  if (launchUrls) {
+    if (launchUrls.length > 1) {
+      return (
+        <MenuLoginWithActionMenu
+          getLoginItems={() => makeSamlAppLoginOptions(launchUrls)}
+          width="100px"
+          onSelect={() => {}} // login item rendered as <a> link for saml apps.
+          buttonText="Log In"
+          size="small"
+          inputType={MenuInputType.NONE}
+          placeholder="Select URL to log in"
+        >
+          {SamlActionMenu}
+        </MenuLoginWithActionMenu>
+      );
+    }
+    ssoUrl = launchUrls[0].url;
+  }
+
+  return (
+    <ButtonWithMenu
+      text="Log In"
+      width="100px"
+      size="small"
+      target="_blank"
+      href={ssoUrl}
+      rel="noreferrer"
+      textTransform="none"
+      forwardedAs="a"
+      title="Log in to SAML application"
+    >
+      {SamlActionMenu}
+    </ButtonWithMenu>
+  );
+}
+
+/**
+ * makeSamlAppLoginButton returns login button for SAML apps.
+ * If launchUrls is not empty and its length is greater than one,
+ * MenuLogin button is returned.
+ * if launchUrls is not empty but its length is exactly one, ButtonBorder
+ * is returned with the href value configured with the launchUrls[0] value.
+ * If launchUrls is empty, ButtonBorder is returned with the href value
+ * configured with samlAppSsoUrl.
+ * @param samlAppSsoUrl - ACS URL (also known as SSO endpoint) value for SAML app.
+ * @param launchUrls - custom service provider endpoints that will be used to initiate
+ *                     authentication, instead of the ACS URL.
+ */
+function makeSamlAppLoginButton(
+  samlAppSsoUrl: string,
+  launchUrls: SamlAppLaunchUrl[]
+) {
+  let ssoUrl = samlAppSsoUrl;
+  if (launchUrls) {
+    if (launchUrls.length > 1) {
+      return (
+        <MenuLogin
+          width="100px"
+          inputType={MenuInputType.NONE}
+          onSelect={() => {}} // login item rendered as <a> link for saml apps.
+          textTransform="none"
+          alignButtonWidthToMenu
+          getLoginItems={() => makeSamlAppLoginOptions(launchUrls)}
+          transformOrigin={{
+            vertical: 'top',
+            horizontal: 'right',
+          }}
+          anchorOrigin={{
+            vertical: 'bottom',
+            horizontal: 'right',
+          }}
+          style={{ borderTopRightRadius: 0, borderBottomRightRadius: 0 }}
```

Comment:
Removing the borders here seems unnecessary, no? There's no action menu to the right of `MenuLogin` here.

Reply:
You are right. removed.